common/async/abort-signals.ts

```diff
@@ -74,3 +74,34 @@ export class BatchedAbortSignals {
     }
   }
 }
+
+/**
+ * A helper class which allows you to use `BatchedAbortSignals` more easily. See the description
+ * of that class for more info.
+ */
+export class RequestCoalescer {
+  private requestsInProgress = new Map<unknown, BatchedAbortSignals>()
+
+  async makeRequest(
+    id: unknown,
```

Comment:
probably this class should be made generic and the generic type should be the ID type, no reason not to have type safety around that